cockatrice/src/dlg_register.cpp

```diff
@@ -30,29 +32,30 @@ DlgRegister::DlgRegister(QWidget *parent) : QDialog(parent)
     portLabel->setBuddy(portEdit);
 
     playernameLabel = new QLabel(tr("Player &name:"));
-    playernameEdit = new QLineEdit(servers.getPlayerName());
+    playernameEdit =
+        new QLineEdit(servers.getPlayerName().isEmpty() ? abstractClient->getUserName() : servers.getPlayerName());
     playernameEdit->setMaxLength(MAX_NAME_LENGTH);
     playernameLabel->setBuddy(playernameEdit);
 
     passwordLabel = new QLabel(tr("P&assword:"));
-    passwordEdit = new QLineEdit();
+    passwordEdit = new QLineEdit(abstractClient->password);
```

Comment:
why instead of all this messing around with accessing the client's private fields just pass the needed information to the dialog's constructor?

Reply:
> why not instead of all this messing around with accessing the client's private fields just pass the needed information to the dialog's constructor?

sure, that would work. Pretty sure I wanted to do it that way as well, cannot remember exactly why I decided to stop and try a different way. I believe we would need to modify the remote client (or the abstract client it inherits from) so that the `window_main` file could access those values. 

So I think in the end we would still need to modify some class to give permission to another class to access its data

A better solution still would be to keep the dialogue open and not destroy it until the user has exited out of the flow by either canceling or quitting the program or successfully registering. I did not do this though because I frankly got intimidated by the code base. It's a little bit complicated. You have to follow how the signals are admitted and then find out where the slots are that are listening for the signals. So like somewhere, there are some signals that cause the dialogue register to close. There's gotta be a failure one and a success one and maybe one or two others for errors or who knows what. We would need to find all of those slots that are listening for those signals and modify them to not close the dialogue if the user encounters a registration error. I also think just hitting the submit button on the registration dialogue close is the registration dialogue, so that one would need to be modified as well. This would definitely be the best approach if I could pull it off though because then we wouldn't need to modify any of the accessibility of the class attributes.

Reply:
I've been meaning to do that yet still haven't found the time to sit down and get it all sorted, it needs a lot of signals and stuff moved https://github.com/Cockatrice/Cockatrice/pull/4586

my comment here is just regarding the arguments in the constructor of the dialog, no changes to the header of the client will be required, just pass the values when the dialog is instantiated.

Reply:
Sounds like an easy change to make then @ebbit1q . RAII is good. I hope to get around to it not too long from now. Thanks for your advice

Reply:
I've changed the constructor to accept arguments: https://github.com/Cockatrice/Cockatrice/pull/4983/files#diff-1b35d27e75120bfb88322ce520f3c057ab93b345875738aba24e04a8596b3285R16